server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java

```diff
@@ -338,6 +341,55 @@ public void invalidAddressesShouldBeAllowed() throws Exception {
     }
 
     @Test
+    public void dateFromHeaderShouldBeUsedIfPresent() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Date: Wed, 17 May 2017 14:18:52 +0200\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZonedDateTime.of(2017, 05, 17, 14, 18, 52, 00, ZoneId.of("Europe/Paris")));
+    }
+
+    @Test
+    public void internalDateShouldBeUsedIfNoDateInHeaders() throws Exception {
+        String headers = "From: user <userdomain>\n"
+            + "To: user1 <user1domain>, user2 <user2domain>\n"
+            + "Cc: usercc <userccdomain>\n"
+            + "Bcc: userbcc <userbccdomain>\n"
+            + "Subject: test subject\n";
+
+        MetaDataWithContent testMail = MetaDataWithContent.builder()
+            .uid(MessageUid.of(2))
+            .flags(new Flags(Flag.SEEN))
+            .size(headers.length())
+            .internalDate(INTERNAL_DATE)
+            .content(new ByteArrayInputStream(headers.getBytes(Charsets.UTF_8)))
+            .attachments(ImmutableList.of())
+            .mailboxId(MAILBOX_ID)
+            .messageId(new TestMessageId.Factory().generate())
+            .build();
+
+        Message testee = messageFactory.fromMetaDataWithContent(testMail);
+
+        assertThat(testee.getDate()).isEqualTo(ZONED_DATE);
+    }
+
```

Comment:
I mean the date is expressed with your timezone defined by your system...

(I deleted other comment about timezone management for the reasons you mention). Here the timezone to use in your test is thus not a hard coded timezone, but the one of the system.